### PR TITLE
Check volume number when completing upgrade

### LINKED ISSUE #240: testing: tests should account for different responses on same endpoint
The sdpctl testing suite should be able to give different responses on multiple requests to the same endpoint.

Currently, the API mocking does not allow for registering multiple responses for the same endpoint, which makes it difficult to test for changing appliance states.
For example, before an upgrade, an appliance has a volume number. Upon a successful upgrade, this volume number will change, indicating that the appliance has switched partition to the new version. In short, if the volume number hasn't changed after `upgrade complete` has finished on the appliance, that would indicate a failed or cancelled upgrade.

The volume number is fetched from the appliance stats API and checking the volume number means that the stats API endpoint need to called at least twice for each appliance that is upgraded. This means testing this scenario is impossible due to the test suite not allowing multiple responses to register for the same endpoint.

### PR DESCRIPTION
**Pull request by kajes**
Add checks for volume number after upgrade completes. If volume number is the same as before the upgrade complete is executed, the upgrade will fail and return the error.

Also fixes breaking tests by adding a new `MutatingResponse` function that allows the test suite to re-use an endpoint, but get different results. The response mutation is done through a callback function provided to the test function.

Fixes #240 and #241

### CODE REVIEW

**Comment by kajes** [comment]
e2e ok https://got-testrail.agi.appgate.com/index.php?/plans/view/1056849

File: pkg/httpmock/http.go [hunk 1]
```diff
@@ -121,6 +121,30 @@ func setup() (*openapi.APIClient, *openapi.Configuration, *http.ServeMux, *httpt
 	return c, clientCfg, mux, server, port, server.Close
 }
 
+func MutatingResponse(filename string, callback func(count int, i []byte) ([]byte, error)) http.HandlerFunc {
+	count := 0
```
Comment (danijeel) [inline pkg/httpmock/http.go:125]: You probably should use a mutex here, I think this will hinder parallel tests using this func.

Comment (kajes) [reply pkg/httpmock/http.go:125]: Probably yes, I'll add it in.

Comment (thomascellerier) [reply pkg/httpmock/http.go:125]: How is the test failing? deadlock?

Comment (kajes) [reply pkg/httpmock/http.go:125]: yep, deadlock.

**Review by danijeel — APPROVED** [review]

**Pull request merged**